lib/clientv2/node.go

```diff
@@ -59,6 +62,63 @@ func (r nodes) Update(ctx context.Context, res *apiv2.Node, opts options.SetOpti
 
 // Delete takes name of the Node and deletes it. Returns an error if one occurs.
 func (r nodes) Delete(ctx context.Context, name string, opts options.DeleteOptions) (*apiv2.Node, error) {
+	pname, err := names.WorkloadEndpointIdentifiers{Node: name}.CalculateWorkloadEndpointName(true)
+	if err != nil {
+		return nil, err
+	}
+
+	// Get all weps belonging to the node
+	weps, err := r.client.WorkloadEndpoints().List(ctx, options.ListOptions{
+		Prefix: true,
+		Name: pname,
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	// Collate all IPs across all endpoints, and then release those IPs.
+	ips := []net.IP{}
+	for _, wep := range weps.Items {
```

Comment:
The prefix match is unfortunately not a perfect match on the Node (since it is theoretically possible for another node to match the prefix (e.g.  a node name of the format `<thisnode>-foobar` would also match a prefix search of the node `<thisnode>`).  Therefore, in this branch and the branch where you delete the WEPs you will also need to check that the Spec.Node field matches the Node.